src/scripts/events/master.js

```diff
@@ -52,89 +51,80 @@ function detachEvents(target) {
  * @description mediates events between master, model and viewModel
  *
  */
-function EventTower(master) {
-
-    if (!(this instanceof EventTower)) {
-        return new EventTower(master);
-    }
+class EventTower {
 
-    this.master = master;
-    this.model = master && master.model || {};
-    this.viewModel = master && master.viewModel || {};
+    constructor(master) {
 
-    //
-    // ensure all targets have an instance of
-    // EventEmitter before proceeding to attach events
-    //
-    if (every([this.master, this.model, this.viewModel], hasEventEmitter)) {
+        this.master = master;
+        this.model = master && master.model || {};
+        this.viewModel = master && master.viewModel || {};
 
         //
-        // attach events to a single instnace
+        // ensure all targets have an instance of
+        // EventEmitter before proceeding to attach events
         //
-        if (!this.master.hasInstance()) {
-            this.attachEvents();
-        }
+        if (every([this.master, this.model, this.viewModel], hasEventEmitter)) {
```

Comment:
@kuhe do you think this check is valid? we control all of those instances, seems like this logic can be moved in to a unit test or something